python_modules/dagster/dagster/_core/definitions/source_asset.py

```diff
@@ -200,3 +277,15 @@ def get_resource_requirements(self) -> Iterator[ResourceRequirement]:
         )
         for source_key, resource_def in self.resource_defs.items():
             yield from resource_def.get_resource_requirements(outer_context=source_key)
+
+    def __hash__(self):
```

Comment:
This strikes me as a tiny bit sketchy. Do we anticipate using source assets as keys in dictionaries?

Reply:
This was added to allow de-duping of source assets inside sets in a downstream PR. Might no longer be required, I'll remove it from this and add it back if it turns out to be needed downsteram.